### PR TITLE
Increase retries for test_merge_tree_azure_blob_storage

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

 Increase max_single_download_retries & max_single_read_retries for test_merge_tree_azure_blob_storage
 
 cc @kssenii @tavplubix

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -1269,13 +1270,15 @@ MergeTreeData::LoadPartResult MergeTreeData::loadDataPart(
         throw;
     }
 #if USE_AZURE_BLOB_STORAGE
-    catch (const Azure::Core::Http::TransportException &)
+    catch (const Azure::Core::RequestFailedException & e)
```
Comment (SmitaRKulkarni) [inline src/Storages/MergeTree/MergeTreeData.cpp:1273]: Updated to catch RequestFailedException azure exception as its the base exception (covers StorageException & TransportException). I think it makes sense to check for this one everywhere & retry on exception

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 2]
```diff
@@ -1269,13 +1270,15 @@ MergeTreeData::LoadPartResult MergeTreeData::loadDataPart(
...
     {
+        LOG_DEBUG(log, "Failed to load data part {}, Exception {}", part_name, e.Message);
```
Comment (SmitaRKulkarni) [inline src/Storages/MergeTree/MergeTreeData.cpp:1275]: We can keep the logs(atleast for now) as they will help to get the issue if part is detached by some other exception

**Comment by tavplubix** [comment]
Do you remember [this](https://github.com/ClickHouse/ClickHouse/pull/53750#discussion_r1304648747) suggestion? Maybe we should do it now

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 3]
```diff
@@ -1244,16 +1244,6 @@ MergeTreeData::LoadPartResult MergeTreeData::loadDataPart(
...
-    {
-        /// Don't count the part as broken if there was a retryalbe error
-        /// during loading, such as "not enough memory" or network error.
-        if (isRetryableException(e))
-            throw;
-
-        mark_broken();
-        return res;
-    }
     catch (const Poco::Net::NetException &)
```
Comment (tavplubix) [inline src/Storages/MergeTree/MergeTreeData.cpp:1247]: Should we add `Poco::Net::NetException` and `Poco::TimeoutException` to `isRetryableException` too?

**Review by tavplubix — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeData.cpp (hunk 3 above)
Comment (SmitaRKulkarni) [reply src/Storages/MergeTree/MergeTreeData.cpp]: addressed it

**Pull request merged**